wagtail/users/views/users.py

```diff
@@ -121,6 +134,18 @@ def get_queryset(self):
         if self.ordering == "username":
             users = users.order_by(User.USERNAME_FIELD)
 
+        if self.is_searching:
+            # Filtering by related fields is not supported in search backend yet
+            # https://docs.wagtail.org/en/stable/topics/search/indexing.html#filtering-on-index-relatedfields
+            if class_is_indexed(User) and not self.group:
+                search_backend = get_search_backend(self.search_backend_name)
+                users = search_backend.search(self.search_query, User.objects.filter())
```

Comment:
We should use the search backend's `autocomplete` method here, rather than `search`. This is because we continuously show search results while the user is typing a search term, and therefore want to show matches on partial words - most of the search backends don't do this for a `search` query (in fact, only the fallback database backend does). This will also mean defining `AutocompleteField` in the user model's `search_fields`.

(Also, I'm not sure what `User.objects.filter()` is doing here - should that be `User.objects.all()`?)